af/fastpath/fastpath/reprocessor.py

```diff
@@ -0,0 +1,322 @@
+            metrics.incr("msmt_count")
+            tar.add(str(msmt_f))
+            measurements.append(msmt_f)
+            tarsize = postcanf.stat().st_size
+            if tarsize > postcan_byte_thresh:
+                log.info(f"Reached {tarsize} bytes")
+                return measurements
+
+    return measurements
+
+
+def parse_date(d):
+    return datetime.strptime(d, "%Y-%m-%d").date()
+
+
+def parse_args():
+    os.environ["TZ"] = "UTC"
+    ap = ArgumentParser(__doc__)
+    ap.add_argument("src_bucket")
+    ap.add_argument("dst_bucket")
+    ap.add_argument("--day", type=lambda d: parse_date(d))
+    ap.add_argument(
+        "--jsonltbl",
+        choices=["dryrun", "insert", "upsert"],
+        default="upsert",
+        help="jsonl table update policy. insert = never overwrite",
+    )
+    ap.add_argument(
+        "--write-fastpath", action="store_true", help="write to fastpath table"
+    )
+    db = "postgresql://shovel:yEqgNr2eXvgG255iEBxVeP@localhost/metadb"
+    ap.add_argument("--db-uri", default=db)
+    ap.add_argument("--s3cachedir", default="s3cachedir")
+    c = ap.parse_args()
+
+    c.no_write_to_db = not c.write_fastpath
+    c.s3cachedir = Path(c.s3cachedir)
+    c.keep_s3_cache = False
+    return c
+
+
+# def score_measurement_and_upsert_fastpath(measurement, msmt_uid) -> None:
+#     raise NotImplementedError
+#     scores = score_measurement(measurement)
+#     anomaly = scores.get("blocking_general", 0.0) > 0.5
+#     failure = scores.get("accuracy", 1.0) < 0.5
+#     confirmed = scores.get("confirmed", False)
+#
+#     sw_name = measurement.get("software_name", "unknown")
+#     sw_version = measurement.get("software_version", "unknown")
+#     platform = "unset"
+#     if "annotations" in measurement and isinstance(measurement["annotations"], dict):
+#         platform = measurement["annotations"].get("platform", "unset")
+#
+#     db.upsert_summary(
+#         measurement,
+#         scores,
+#         anomaly,
+#         confirmed,
+#         failure,
+#         msmt_uid,
+#         sw_name,
+#         sw_version,
+#         platform,
+#         conf.update,
+#     )
+
+
+@dataclass
+class Entity:
+    jsonlf: Path
+    fd: int
+    jsonl_s3path: str
+    lookup_list: list
+
+
+seen_uids = set()
+
+
+def finalize_jsonl(s3sig, db_conn, conf, e: Entity) -> None:
+    jsize = int(e.fd.offset / 1024)
+    log.info(f"Closing jsonl and uploading it. Size: {jsize} KB")
+    e.fd.close()
+    upload_to_s3(s3sig, conf.dst_bucket, e.jsonlf, e.jsonl_s3path)
+    update_db_table(db_conn, e.lookup_list, conf.jsonltbl)
+    e.jsonlf.unlink()
+
+
+def process_measurement(msm_tup, buf, conf, s3sig, db_conn):
+    """Process a msmt
+    If needed: create a new Entity tracking a jsonl file,
+      close and upload jsonl to S3 and upsert db
+    """
+    THRESHOLD = 20 * 1024 * 1024
+    msm_jstr, msm, _ = msm_tup
+    if msm is None:
+        msm = ujson.loads(msm_jstr)
+    # TODO: yaml?
```

Comment:
From what I gather reading the code of `load_multiple` when the format is in YAML you should always be getting a non null msm value, so I guess there is no need to handle YAML normalization and parsing in here.

Reply:
Yes, It's updated now.